File: app/src/main/java/kr/djspi/pipe01/dto/SpiType.java
package kr.djspi.pipe01.dto;

import org.jetbrains.annotations.Nullable;

import java.io.Serializable;

import lombok.Data;
import lombok.Getter;

@Data
public class SpiType implements DataItem, Serializable {

    private final int id;
    /**
     * SPI 제품 종류: 표지기, 표지판, 표지주
     */
    private final String type;

    @Getter
    @SuppressWarnings("NonAsciiCharacters")
    public enum SpiTypeEnum {

        표지판("표지판", "plate"),
        표지기("표지기", "marker"),
        표지주("표지주", "column");

        private String name;
        private String code;

        SpiTypeEnum(String name, String code) {
            this.name = name;
            this.code = code;
        }

        @Nullable
        public static String parseSpiType(String spiType) {
            for (SpiTypeEnum typeEnum : SpiTypeEnum.values()) {
                if (typeEnum.name.equals(spiType)) {
                    return typeEnum.getCode();
                }
            }
            return null;
        }
    }
}

File: app/src/main/java/kr/djspi/pipe01/dto/SpiType.kt
package kr.djspi.pipe01.dto

import java.io.Serializable

/**
 * SPI 제품 종류: 표지기, 표지판, 표지주
 */
class SpiType(val id: Int = -1, val type: String) : DataItem, Serializable {

    @Suppress("NonAsciiCharacters", "EnumEntryName", "unused")
    enum class SpiTypeEnum(internal var typeName: String, internal var code: String) {

        표지판("표지판", "plate"),
        표지기("표지기", "marker"),
        표지주("표지주", "column");

        companion object {

            fun parseSpiType(spiType: String?): String? {
                values().forEach {
                    if (it.typeName == spiType) {
                        return it.code
                    }
                }
                return null
            }
        }
    }
}
